packages/foreman/foreman-fapolicyd/foreman-fapolicyd.spec

```diff
@@ -29,11 +32,17 @@ cp 61-foreman-proxy.rules %{buildroot}%{_sysconfdir}/fapolicyd/rules.d/61-forema
 %post
 
 %{_sbindir}/fagenrules --load
+%{_bindir}/systemctl try-restart fapolicyd.service
 
 %postun
 
 %{_sbindir}/fagenrules --load
 
+if %{_bindir}/systemctl is-active --quiet fapolicyd.service; then
```

Comment:
I can't tell if this is still needed or not for the postun? I would assume the macro handles it.

Reply:
On EL9:
```
%systemd_postun_with_restart() \
%{expand:%%{?__systemd_someargs_%#:%%__systemd_someargs_%# systemd_postun_with_restart}} \
if [ $1 -ge 1 ] && [ -x "/usr/lib/systemd/systemd-update-helper" ]; then \
    # Package upgrade, not uninstall \
    /usr/lib/systemd/systemd-update-helper mark-restart-system-units %{?*} || : \
fi \
%{nil}
```

From `/usr/lib/systemd/systemd-update-helper`
```sh
    mark-restart-system-units)
        [ -d /run/systemd/system ] || exit 0

        for unit in "$@"; do
            systemctl set-property "$unit" Markers=+needs-restart &
        done
        wait
        ;;
```

And I recall that at the end of the RPM transaction it will look for all running services with the `needs-restart` marker. This prevents multiple packages restarting the same service.

On EL8 we have the more classic implementation:
```
%systemd_postun_with_restart() \
if [ $1 -ge 1 ] ; then \
        # Package upgrade, not uninstall \
        systemctl try-restart %{?*} &>/dev/null || : \
fi \
%{nil}
```

So I think it's redundant.